tests/dataset_builders/pie/test_cdcp.py

```diff
@@ -0,0 +1,431 @@
+
+
+@pytest.fixture(scope="module")
+def generate_document_kwargs(hf_dataset, split):
+    return {
+        "relation_int2str": hf_dataset[split].features["relations"].feature["label"].int2str,
+        "proposition_int2str": hf_dataset[split].features["propositions"].feature["label"].int2str,
+    }
+
+
+@pytest.fixture(scope="module")
+def convert_back_kwargs(hf_dataset, split):
+    return {
+        "relation_str2int": hf_dataset[split].features["relations"].feature["label"].str2int,
+        "proposition_str2int": hf_dataset[split].features["propositions"].feature["label"].str2int,
+    }
+
+
+def test_example_to_document(hf_example, generate_document_kwargs):
+    doc = example_to_document(hf_example, **generate_document_kwargs)
+    assert doc is not None
+
+
+def test_example_to_document_and_back(hf_example, generate_document_kwargs, convert_back_kwargs):
+    doc = example_to_document(hf_example, **generate_document_kwargs)
+    hf_example_back = document_to_example(doc, **convert_back_kwargs)
+    _deep_compare(
+        obj=hf_example_back,
+        obj_expected=hf_example,
+    )
+
+
+def _assert_no_span_overlap(document: Document, text_field: str, span_layer: str):
+    spans = document[span_layer]
+    text = getattr(document, text_field)
+    seq = [None] * len(text)
+    for span in spans:
+        assert seq[span.start : span.end] == [None] * len(text[span.start : span.end])
+        seq[span.start : span.end] = text[span.start : span.end]
+
+
+def test_assert_no_span_overlap():
+    @dataclasses.dataclass
+    class TextDocumentWithEntities(TextBasedDocument):
+        entities: AnnotationList[LabeledSpan] = annotation_field(target="text")
+
+    doc0 = TextDocumentWithEntities(text="abcdefghijklmnop")
+    doc0.entities.append(LabeledSpan(start=0, end=4, label="A"))
+    doc0.entities.append(LabeledSpan(start=4, end=6, label="B"))
+
+    # this should work
+    _assert_no_span_overlap(document=doc0, text_field="text", span_layer="entities")
+
+    doc1 = TextDocumentWithEntities(text="abcdefghijklmnop")
+    doc1.entities.append(LabeledSpan(start=0, end=4, label="A"))
+    doc1.entities.append(LabeledSpan(start=2, end=6, label="B"))
+
+    # this should fail
+    with pytest.raises(AssertionError):
+        _assert_no_span_overlap(document=doc1, text_field="text", span_layer="entities")
+
+
+def test_example_to_document_and_back_all(
+    hf_dataset, generate_document_kwargs, convert_back_kwargs, split
+):
+    for hf_ex in hf_dataset[split]:
+        doc = example_to_document(hf_ex, **generate_document_kwargs)
+        _assert_no_span_overlap(document=doc, text_field="text", span_layer="propositions")
+        hf_example_back = document_to_example(doc, **convert_back_kwargs)
+        _deep_compare(
+            obj=hf_example_back,
+            obj_expected=hf_ex,
+        )
+
+
+@pytest.fixture(scope="module")
+def dataset() -> DatasetDict:
+    return DatasetDict.load_dataset(PIE_DATASET_PATH)

```

Comment:
Since `PIE_DATASET_PATH` should be now a `Path` object (see above comment), we need convert it back to a string to work with that method:
```suggestion
    return DatasetDict.load_dataset(str(PIE_DATASET_PATH))

```

Reply:
fix in ec6b346